rspace/src/test/scala/coop/rchain/rspace/HotStoreSpec.scala

```diff
@@ -503,6 +507,76 @@ trait HotStoreSpec[F[_], M[_]] extends FlatSpec with Matchers with GeneratorDriv
       }
   }
 
+  "installJoin" should "cache installed joins separately" in forAll {
+    (
+        channel: Channel,
+        cachedJoins: Joins,
+        insertedJoin: Join,
+        installedJoin: Join
+    ) =>
+      whenever(insertedJoin != installedJoin) {
+        fixture { (state, history, hotStore) =>
+          {
+            for {
+              _ <- state.modify(
+                    _ =>
+                      Cache(
+                        joins = TrieMap(
+                          channel -> cachedJoins
+                        )
+                      )
+                  )
+              _     <- hotStore.putJoin(channel, insertedJoin)
+              _     <- hotStore.installJoin(channel, installedJoin)
+              cache <- state.read
+              _ <- S.delay {
+                    cache
+                      .installedJoins(channel) shouldEqual Seq(installedJoin)
+                  }
+              _ <- S.delay(
+                    cache
+                      .joins(channel) shouldEqual insertedJoin +: cachedJoins
+                  )
+            } yield ()
+          }
+        }
+      }
+  }
+
+  it should "allow installing multiple values per channel" in forAll {
```

Comment:
do you feel that this is an exhaustive test? maybe it would be beneficial to accept as generated param a list of joins and put them in a loop? testing exactly two joins feels wrong.